github.com/cockroachdb/cockroach/pkg/storage: TestReplicaLease failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/14236e0225a7ac420c563a6e794b022a619adc15

Parameters:
```
TAGS=
GOFLAGS=
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=670026&tab=buildLog

```

E180521 06:08:02.697138 12502 vendor/github.com/coreos/etcd/raft/raft.go:1383  [s1,r1/1:/M{in-ax}] 1 state.commit 15 is out of range [10, 14]
panic: 1 state.commit 15 is out of range [10, 14] [recovered]
	panic: 1 state.commit 15 is out of range [10, 14]

goroutine 12502 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc420646480, 0x26f5320, 0xc420a01470)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:178 +0x11f
panic(0x20236c0, 0xc42054efa0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
github.com/cockroachdb/cockroach/pkg/storage.(*raftLogger).Panicf(0xc42054ef70, 0x23ad6fa, 0x2b, 0xc4203f9780, 0x4, 0x4)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/raft.go:106 +0xfe
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.(*raft).loadState(0xc420834b00, 0x6, 0x1, 0xf, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:1383 +0x1c7
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.newRaft(0xc420d85628, 0x235b7e1)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/raft.go:352 +0xd40
github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft.NewRawNode(0xc420d85628, 0x0, 0x0, 0x0, 0xc420a01110, 0xbeb8b670a4dff94b, 0x19f1c0ebc)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/coreos/etcd/raft/rawnode.go:79 +0x71
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).withRaftGroupLocked(0xc42055f180, 0x248bcd0, 0xc42055f278, 0xc420a01470)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:545 +0x2aa
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).maybeInitializeRaftGroup(0xc42055f180, 0x26f5320, 0xc420a01470)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1820 +0x11b
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send(0xc42055f180, 0x26f5320, 0xc420a01470, 0x35a4e991, 0x14, 0x0, 0x0, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1844 +0x136
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).requestLeaseAsync.func2(0x26f5320, 0xc420a01470)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:310 +0x5af
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc420646480, 0x26f5320, 0xc420a01470, 0xc4207b1100, 0x35, 0x2719040, 0xc42045c7e0, 0xc420d9aa20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xe6
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:259 +0x133

```

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -804,7 +804,10 @@ func TestReplicaLease(t *testing.T) {
 	tc := testContext{}
 	stopper := stop.NewStopper()
 	defer stopper.Stop(context.TODO())
-	tc.Start(t, stopper)
+	tc.manualClock = hlc.NewManualClock(123)
+	tsc := TestStoreConfig(hlc.NewClock(tc.manualClock.UnixNano, time.Nanosecond))
+	tsc.TestingKnobs.DisableAutomaticLeaseRenewal = true
+	tc.StartWithStoreConfig(t, stopper, tsc)
 	secondReplica, err := tc.addBogusReplicaToRangeDesc(context.TODO())
 	if err != nil {
 		t.Fatal(err)
@@ -858,21 +861,16 @@ func TestReplicaLease(t *testing.T) {
 	}
 
 	// Verify that command returns NotLeaseHolderError when lease is rejected.
-	repl, err := NewReplica(testRangeDescriptor(), tc.store, 0)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	repl.mu.Lock()
-	repl.mu.submitProposalFn = func(*ProposalData) error {
+	tc.repl.mu.Lock()
+	tc.repl.mu.submitProposalFn = func(*ProposalData) error {
 		return &roachpb.LeaseRejectedError{
 			Message: "replica not found",
 		}
 	}
-	repl.mu.Unlock()
+	tc.repl.mu.Unlock()
 
 	{
-		_, err := repl.redirectOnOrAcquireLease(context.Background())
+		_, err := tc.repl.redirectOnOrAcquireLease(context.Background())
 		if _, ok := err.GetDetail().(*roachpb.NotLeaseHolderError); !ok {
 			t.Fatalf("expected %T, got %s", &roachpb.NotLeaseHolderError{}, err)
 		}

